Util/channel/src/main/java/io/deephaven/util/channel/SeekableChannelsProvider.java

```diff
@@ -85,4 +86,24 @@ default SeekableByteChannel getWriteChannel(@NotNull final String path, final bo
     }
 
     SeekableByteChannel getWriteChannel(@NotNull Path path, boolean append) throws IOException;
+
+    /**
+     * Applies the supplied action on the list of file URIs contained inside the given directory. The logic is
+     * non-recursive.
+     *
+     * @param directoryURI The URI of the directory to list
+     * @param processor A method to apply to the URIs in the directory
+     */
+    void applyToChildURIs(@NotNull URI directoryURI, @NotNull Consumer<URI> processor) throws IOException;
```

Comment:
Had a brief discussion with Ryan, he agreed that this seems like the best place to keep these methods since Provider is kind of like out interface into the filesystem. Please let me know if you have any suggestions for better function names or where to place them.